packages/engine/paima-sm/src/index.ts

```diff
@@ -462,31 +515,50 @@ async function processScheduledData(
 
       // Trigger STF
       let sqlQueries: SQLUpdate[] = [];
+      let eventsToEmit: EventsToEmit<Events[string][number]> = [];
       try {
-        sqlQueries = await gameStateTransition(
+        const { stateTransitions, events } = await gameStateTransition(
           inputData,
           { blockHeight: data.block_height, timestamp: latestChainData.timestamp },
           randomnessGenerator,
           DBConn
         );
+
+        sqlQueries = stateTransitions;
+
+        handleEvents<Events[string][number]>(
+          events,
+          sqlQueries,
+          txIndexInBlock,
+          latestChainData,
+          eventDefinitions,
+          eventsToEmit
+        );
       } catch (err) {
         // skip scheduled data where the STF fails
         doLog(`[paima-sm] Error on scheduled data STF call. Skipping`, err);
         continue;
       }
       if (sqlQueries.length !== 0) {
-        await tryOrRollback(DBConn, async () => {
+        const success = await tryOrRollback(DBConn, async () => {
```

Comment:
I don't think it's ever possible for `success` to be false here, given an error goes into the `catch` block

Reply:
Not false, but it should be `undefined` if it goes into the `catch` block, which is falsy. In that case it won't enter the if branch, which is the only thing that matters. Technically this could be a singleton object, but not sure there is a convenient one. Or we could change the `tryOrRollback` function to return `false` I guess, but then it wouldn't be generic on the function type.